go/cache/lru_cache_test.go

```diff
@@ -27,24 +29,13 @@ type CacheValue struct {
 func TestInitialState(t *testing.T) {
 	cache := NewLRUCache[*CacheValue](5)
 	l, sz, c, e, h, m := cache.Len(), cache.UsedCapacity(), cache.MaxCapacity(), cache.Evictions(), cache.Hits(), cache.Misses()
-	if l != 0 {
-		t.Errorf("length = %v, want 0", l)
-	}
-	if sz != 0 {
-		t.Errorf("size = %v, want 0", sz)
-	}
-	if c != 5 {
-		t.Errorf("capacity = %v, want 5", c)
-	}
-	if e != 0 {
-		t.Errorf("evictions = %v, want 0", c)
-	}
-	if h != 0 {
-		t.Errorf("hits = %v, want 0", c)
-	}
-	if m != 0 {
-		t.Errorf("misses = %v, want 0", c)
-	}
+	assert.Equal(t, 0, l, "length = %v, want 0", l)
+	assert.Equal(t, int64(0), sz, "size = %v, want 0", sz)
+	assert.Equal(t, int64(5), c, "capacity = %v, want 5", c)
```

Comment:
When using the `asset` I don't think it is useful to add the expectations in the message field. The method already prints the wanted value and the received value:
```
Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestInitialState
        	Messages:   	size = 0, want 1
```

This apply to all the files you modified in this PR

Reply:
done, I'll wait on the the CI checks tho